---
title: Proteomics Data Analysis 2021 (PDA21)
output:
    html_document:
      theme: cosmo      
      code_download: false
      toc: false
      toc_float: false
      number_sections: false

---


```{r setup, include=FALSE, cache=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

***

```{r echo=FALSE}
knitr::include_graphics("./figures/IntroFig.png")
```


### Course Description

Mass spectrometry based proteomic experiments generate ever larger datasets and, as a consequence, complex data interpretation challenges. This course focuses on the statistical concepts for peptide identification, quantification, and differential analysis. Moreover, more advanced experimental designs and blocking will also be introduced. The core focus will be on shotgun proteomics data, and quantification using label-free precursor peptide (MS1) ion intensities. The course will rely exclusively on free and userfriendly opensource tools in R/Bioconductor. The course will provide a solid basis for beginners, but will also bring new perspectives to those already familiar with standard data interpretation procedures in proteomics.

Students can sharpen their background knowledge on Mass Spectrometry, Proteomics & Bioinformatics for Proteomics
 here:
[Mass Spectrometry and Bioinformatics for Proteomics](./techvid.html)

### Target Audience
This course is oriented towards biologists and bioinformaticians with a particular interest in differential analysis for quantitative proteomics.

According to the target audience of the course we either work with a graphical user interface (GUI) in a R/shiny App msqrob2gui e.g. 
- Proteomics Bioinformatics course of the EBI
- Proteomics Data Analysis course at the Gulbenkian institute) 
- BePA introduction: What can proteomics do for you?

or with R/markdowns scripts e.g. 

- Bioinformatics Summer School at UCLouvain 
- Statistical Genomics Course at Ghent University. 

### Software & Data 

- [Install msqrob2 software](./software.html)
- [Download data](https://github.com/statOmics/pda/archive/refs/heads/data.zip)

Note, that users who develop R/markdown scripts can access data both from the web or from disk within their scripts. So they do not need to download the data first. The msqrob2gui Shiny App only works with data that is available on disk.  


### Issues

If you encounter problems related to the course material (e.g. package installation problems, bugs in the code, typos, ...), please consider [posting an issue on GitHub](https://github.com/statOmics/PDA21/issues).

---

### Detailed Program


  1. Preprocessing & Analysis of Label Free Quantitative Proteomics Experiments with Simple Designs
  - Lecture: [Preprocessing](./pda_quantification_preprocessing.html), [[PDF](./pda_quantification_preprocessing.pdf)]
  - Tutorial: [preprocessing](./pda_tutorialPreprocessing.html)

  2. Statistical Inference & Analysis of Experiments with Factorial Designs
  - Lecture: [Differential Abundance Analysis](./pda_quantification_inference.html), [[PDF](./pda_quantification_inference.pdf)]
  - Tutorial: [Design](da_tutorialDesign.html)

---

All materials are licensed under the [Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International (CC BY-NC-SA 4.0)](https://creativecommons.org/licenses/by-nc-sa/4.0)

