<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Statistical Methods for Quantitative MS-based Proteomics: Peptide-level Models for Summarization and Inference</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PDA21</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="pda_quantification_preprocessing.html">Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">Wrap-up DA - Blocking</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="pda_tutorialPreprocessing.html">Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">Quantification - Design</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statOmics/PDA21">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Statistical Methods for Quantitative MS-based Proteomics: Peptide-level Models for Summarization and Inference</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date"><a href="https://statomics.github.io">statOmics</a>, Ghent University</h4>

</div>


<p><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a></p>
<p>This is part of the online course <a href="https://statomics.github.io/PDA21/">Proteomics Data Analysis 2021 (PDA21)</a></p>
<div id="import-the-data-in-r" class="section level1">
<h1><span class="header-section-number">1</span> Import the data in R</h1>
<div id="load-libraries" class="section level2">
<h2><span class="header-section-number">1.1</span> Load libraries</h2>
<details>
<summary> Click to see code </summary>
<p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">library</span>(limma)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw">library</span>(QFeatures)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="kw">library</span>(msqrob2)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="kw">library</span>(plotly)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="kw">library</span>(gridExtra)</span></code></pre></div>
</p>
</details>
</div>
<div id="read-data" class="section level2">
<h2><span class="header-section-number">1.2</span> Read data</h2>
<details>
<summary> Click to see background and code </summary>
<p>
<ol style="list-style-type: decimal">
<li>We use a peptides.txt file from MS-data quantified with maxquant that contains MS1 intensities summarized at the peptide level.</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>peptidesFile &lt;-<span class="st"> &quot;https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/fullCptacDatasSetNotForTutorial/peptides.txt&quot;</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Maxquant stores the intensity data for the different samples in columnns that start with Intensity. We can retreive the column names with the intensity data with the code below:</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>ecols &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;Intensity</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="kw">names</span>(<span class="kw">read.delim</span>(peptidesFile)))</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Read the data and store it in QFeatures object</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>pe &lt;-<span class="st"> </span><span class="kw">readQFeatures</span>(</span>
<span id="cb4-2"><a href="#cb4-2"></a>  <span class="dt">table =</span> peptidesFile,</span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="dt">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="dt">ecol =</span> ecols,</span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="dt">name =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="design" class="section level2">
<h2><span class="header-section-number">1.3</span> Design</h2>
<details>
<summary> Click to see background and code </summary>
<p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>pe <span class="op">%&gt;%</span><span class="st"> </span>colnames</span></code></pre></div>
<pre><code>## CharacterList of length 1
## [[&quot;peptideRaw&quot;]] Intensity.6A_1 Intensity.6A_2 ... Intensity.6E_9</code></pre>
<ul>
<li><p>Note, that the sample names include the spike-in condition.</p></li>
<li><p>They also end on a number.</p>
<ul>
<li>1-3 is from lab 1,</li>
<li>4-6 from lab 2 and</li>
<li>7-9 from lab 3.</li>
</ul></li>
<li><p>We update the colData with information on the design</p></li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw">colData</span>(pe)<span class="op">$</span>lab &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="kw">rep</span>(<span class="kw">paste0</span>(<span class="st">&quot;lab&quot;</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>),<span class="dt">each=</span><span class="dv">3</span>),<span class="dv">5</span>) <span class="op">%&gt;%</span><span class="st"> </span>as.factor</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="kw">colData</span>(pe)<span class="op">$</span>condition &lt;-<span class="st"> </span>pe[[<span class="st">&quot;peptideRaw&quot;</span>]] <span class="op">%&gt;%</span><span class="st"> </span>colnames <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">substr</span>(<span class="dv">12</span>,<span class="dv">12</span>) <span class="op">%&gt;%</span><span class="st"> </span>as.factor</span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="kw">colData</span>(pe)<span class="op">$</span>spikeConcentration &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="kw">c</span>(<span class="dt">A =</span> <span class="fl">0.25</span>, <span class="dt">B =</span> <span class="fl">0.74</span>, <span class="dt">C =</span> <span class="fl">2.22</span>, <span class="dt">D =</span> <span class="fl">6.67</span>, <span class="dt">E =</span> <span class="dv">20</span>),<span class="dt">each =</span> <span class="dv">9</span>)</span></code></pre></div>
<ul>
<li>We explore the colData</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="kw">colData</span>(pe)</span></code></pre></div>
<pre><code>## DataFrame with 45 rows and 3 columns
##                     lab condition spikeConcentration
##                &lt;factor&gt;  &lt;factor&gt;          &lt;numeric&gt;
## Intensity.6A_1     lab1         A               0.25
## Intensity.6A_2     lab1         A               0.25
## Intensity.6A_3     lab1         A               0.25
## Intensity.6A_4     lab2         A               0.25
## Intensity.6A_5     lab2         A               0.25
## ...                 ...       ...                ...
## Intensity.6E_5     lab2         E                 20
## Intensity.6E_6     lab2         E                 20
## Intensity.6E_7     lab3         E                 20
## Intensity.6E_8     lab3         E                 20
## Intensity.6E_9     lab3         E                 20</code></pre>
</p>
</details>
</div>
<div id="preprocessing" class="section level2">
<h2><span class="header-section-number">1.4</span> Preprocessing</h2>
<div id="log-transform" class="section level3">
<h3><span class="header-section-number">1.4.1</span> Log-transform</h3>
<details>
<summary> Click to see code to log-transfrom the data </summary>
<p>
<ul>
<li>We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="op">$</span>nNonZero &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">assay</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</span></code></pre></div>
<ul>
<li>Peptides with zero intensities are missing peptides and should be represent with a <code>NA</code> value rather than <code>0</code>.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>pe &lt;-<span class="st"> </span><span class="kw">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<ul>
<li>Logtransform data with base 2</li>
</ul>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>pe &lt;-<span class="st"> </span><span class="kw">logTransform</span>(pe, <span class="dt">base =</span> <span class="dv">2</span>, <span class="dt">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="filtering" class="section level3">
<h3><span class="header-section-number">1.4.2</span> Filtering</h3>
<details>
<summary> Click to see code to filter the data </summary>
<p>
<ol style="list-style-type: decimal">
<li>Handling overlapping protein groups</li>
</ol>
<p>In our approach a peptide can map to multiple proteins, as long as there is none of these proteins present in a smaller subgroup.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>pe[[<span class="st">&quot;peptideLog&quot;</span>]] &lt;-</span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="st"> </span>pe[[<span class="st">&quot;peptideLog&quot;</span>]][<span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span>Proteins</span>
<span id="cb13-3"><a href="#cb13-3"></a> <span class="op">%in%</span><span class="st"> </span><span class="kw">smallestUniqueGroups</span>(<span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span>Proteins),]</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Remove reverse sequences (decoys) and contaminants</li>
</ol>
<p>We now remove the contaminants, peptides that map to decoy sequences, and proteins which were only identified by peptides with modifications.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>pe[[<span class="st">&quot;peptideLog&quot;</span>]] &lt;-<span class="st"> </span>pe[[<span class="st">&quot;peptideLog&quot;</span>]][<span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span>Reverse <span class="op">!=</span><span class="st"> &quot;+&quot;</span>, ]</span>
<span id="cb14-2"><a href="#cb14-2"></a>pe[[<span class="st">&quot;peptideLog&quot;</span>]] &lt;-<span class="st"> </span>pe[[<span class="st">&quot;peptideLog&quot;</span>]][<span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span></span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="st">    </span>Potential.contaminant <span class="op">!=</span><span class="st"> &quot;+&quot;</span>, ]</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Drop peptides that were only identified in one sample</li>
</ol>
<p>We keep peptides that were observed at last twice.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>pe[[<span class="st">&quot;peptideLog&quot;</span>]] &lt;-<span class="st"> </span>pe[[<span class="st">&quot;peptideLog&quot;</span>]][<span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="op">$</span>nNonZero <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>, ]</span>
<span id="cb15-2"><a href="#cb15-2"></a><span class="kw">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 10478</code></pre>
We keep 10478 peptides upon filtering.
</p>
</details>
</div>
</div>
<div id="normalization" class="section level2">
<h2><span class="header-section-number">1.5</span> Normalization</h2>
<details>
<summary> Click to see R-code to normalize the data </summary>
<p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>pe &lt;-<span class="st"> </span><span class="kw">normalize</span>(pe, </span>
<span id="cb17-2"><a href="#cb17-2"></a>                <span class="dt">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb17-3"><a href="#cb17-3"></a>                <span class="dt">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb17-4"><a href="#cb17-4"></a>                <span class="dt">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
</p>
</details>
<hr />
</div>
</div>
<div id="peptide-level-models" class="section level1">
<h1><span class="header-section-number">2</span> Peptide-level models</h1>
<div id="summarization" class="section level2">
<h2><span class="header-section-number">2.1</span> Summarization</h2>
<details>
<summary> Click to see code to make plot </summary>
<p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>prot &lt;-<span class="st"> &quot;P01031ups|CO5_HUMAN_UPS&quot;</span></span>
<span id="cb18-2"><a href="#cb18-2"></a>data &lt;-<span class="st"> </span>pe[[<span class="st">&quot;peptideNorm&quot;</span>]][</span>
<span id="cb18-3"><a href="#cb18-3"></a>  <span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideNorm&quot;</span>]])<span class="op">$</span>Proteins <span class="op">==</span><span class="st"> </span>prot,</span>
<span id="cb18-4"><a href="#cb18-4"></a>  <span class="kw">colData</span>(pe)<span class="op">$</span>lab<span class="op">==</span><span class="st">&quot;lab3&quot;</span>] <span class="op">%&gt;%</span></span>
<span id="cb18-5"><a href="#cb18-5"></a><span class="st">  </span>assay <span class="op">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6"></a><span class="st">  </span>as.data.frame <span class="op">%&gt;%</span></span>
<span id="cb18-7"><a href="#cb18-7"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;peptide&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb18-8"><a href="#cb18-8"></a><span class="st">  </span><span class="kw">gather</span>(sample, intensity, <span class="op">-</span>peptide) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-9"><a href="#cb18-9"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">condition =</span> <span class="kw">colData</span>(pe)[sample,<span class="st">&quot;condition&quot;</span>]) <span class="op">%&gt;%</span></span>
<span id="cb18-10"><a href="#cb18-10"></a><span class="st">  </span>na.exclude</span>
<span id="cb18-11"><a href="#cb18-11"></a>sumPlot &lt;-<span class="st"> </span>data <span class="op">%&gt;%</span></span>
<span id="cb18-12"><a href="#cb18-12"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> peptide, <span class="dt">y =</span> intensity, <span class="dt">color =</span> condition, <span class="dt">group =</span> sample, <span class="dt">label =</span> condition), <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb18-13"><a href="#cb18-13"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb18-14"><a href="#cb18-14"></a><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span></span>
<span id="cb18-15"><a href="#cb18-15"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust =</span> <span class="dv">1</span>)) <span class="op">+</span></span>
<span id="cb18-16"><a href="#cb18-16"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Peptide&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb18-17"><a href="#cb18-17"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Intensity (log2)&quot;</span>) <span class="op">+</span></span>
<span id="cb18-18"><a href="#cb18-18"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;protein: &quot;</span>,prot))</span></code></pre></div>
</p>
</details>
<p>Here, we will focus on the summarization of the intensities for protein P01031ups|CO5_HUMAN_UPS.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>sumPlot</span></code></pre></div>
<p><img src="pda_robustSummarisation_peptideModels_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<div id="median-summarization" class="section level3">
<h3><span class="header-section-number">2.1.1</span> Median summarization</h3>
<p>We first evaluate median summarization for protein P01031ups|CO5_HUMAN_UPS.</p>
<details>
<summary> Click to see code to make plot </summary>
<p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>dataHlp &lt;-<span class="st"> </span>pe[[<span class="st">&quot;peptideNorm&quot;</span>]][</span>
<span id="cb20-2"><a href="#cb20-2"></a>    <span class="kw">rowData</span>(pe[[<span class="st">&quot;peptideNorm&quot;</span>]])<span class="op">$</span>Proteins <span class="op">==</span><span class="st"> </span>prot,</span>
<span id="cb20-3"><a href="#cb20-3"></a>    <span class="kw">colData</span>(pe)<span class="op">$</span>lab<span class="op">==</span><span class="st">&quot;lab3&quot;</span>] <span class="op">%&gt;%</span><span class="st"> </span>assay </span>
<span id="cb20-4"><a href="#cb20-4"></a></span>
<span id="cb20-5"><a href="#cb20-5"></a>sumMedian &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</span>
<span id="cb20-6"><a href="#cb20-6"></a>  <span class="dt">intensity=</span> dataHlp</span>
<span id="cb20-7"><a href="#cb20-7"></a>    <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">colMedians</span>(<span class="dt">na.rm=</span><span class="ot">TRUE</span>)</span>
<span id="cb20-8"><a href="#cb20-8"></a>  ,</span>
<span id="cb20-9"><a href="#cb20-9"></a>  <span class="dt">condition=</span> <span class="kw">colnames</span>(dataHlp) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">substr</span>(<span class="dv">12</span>,<span class="dv">12</span>) <span class="op">%&gt;%</span><span class="st"> </span>as.factor )</span>
<span id="cb20-10"><a href="#cb20-10"></a></span>
<span id="cb20-11"><a href="#cb20-11"></a>sumMedianPlot &lt;-<span class="st"> </span>sumPlot <span class="op">+</span><span class="st"> </span></span>
<span id="cb20-12"><a href="#cb20-12"></a><span class="st">  </span><span class="kw">geom_hline</span>(</span>
<span id="cb20-13"><a href="#cb20-13"></a>    <span class="dt">data =</span> sumMedian,</span>
<span id="cb20-14"><a href="#cb20-14"></a>    <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">yintercept=</span>intensity,<span class="dt">color=</span>condition)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb20-15"><a href="#cb20-15"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Median summarization&quot;</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>sumMedianPlot</span></code></pre></div>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_hline).</code></pre>
<p><img src="pda_robustSummarisation_peptideModels_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<ul>
<li>The sample medians are not a good estimate for the protein expression value.</li>
<li>Indeed, they do not account for differences in peptide effects</li>
<li>Peptides that ionize poorly are also picked up in samples with high spike-in concencentration and not in samples with low spike-in concentration</li>
<li>This introduces a bias.</li>
</ul>
</div>
<div id="model-based-summarization" class="section level3">
<h3><span class="header-section-number">2.1.2</span> Model based summarization</h3>
<p>We can use a linear peptide-level model to estimate the protein expression value while correcting for the peptide effect, i.e. </p>
<p><span class="math display">\[ 
y_{ip} = \beta_i^\text{sample}+\beta^{peptide}_{p} + \epsilon_{ip}
\]</span></p>
<details>
<summary> Click to see code to make plot </summary>
<p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>sumMeanPepMod &lt;-<span class="st"> </span><span class="kw">lm</span>(intensity <span class="op">~</span><span class="st"> </span><span class="dv">-1</span> <span class="op">+</span><span class="st"> </span>sample <span class="op">+</span><span class="st"> </span>peptide,data)</span>
<span id="cb23-2"><a href="#cb23-2"></a></span>
<span id="cb23-3"><a href="#cb23-3"></a>sumMeanPep &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</span>
<span id="cb23-4"><a href="#cb23-4"></a>  <span class="dt">intensity=</span>sumMeanPepMod<span class="op">$</span>coef[<span class="kw">grep</span>(<span class="st">&quot;sample&quot;</span>,<span class="kw">names</span>(sumMeanPepMod<span class="op">$</span>coef))] <span class="op">+</span><span class="st"> </span><span class="kw">mean</span>(data<span class="op">$</span>intensity) <span class="op">-</span><span class="st"> </span><span class="kw">mean</span>(sumMeanPepMod<span class="op">$</span>coef[<span class="kw">grep</span>(<span class="st">&quot;sample&quot;</span>,<span class="kw">names</span>(sumMeanPepMod<span class="op">$</span>coef))]),</span>
<span id="cb23-5"><a href="#cb23-5"></a>  <span class="dt">condition=</span> <span class="kw">names</span>(sumMeanPepMod<span class="op">$</span>coef)[<span class="kw">grep</span>(<span class="st">&quot;sample&quot;</span>,<span class="kw">names</span>(sumMeanPepMod<span class="op">$</span>coef))] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">substr</span>(<span class="dv">18</span>,<span class="dv">18</span>) <span class="op">%&gt;%</span><span class="st"> </span>as.factor )</span>
<span id="cb23-6"><a href="#cb23-6"></a></span>
<span id="cb23-7"><a href="#cb23-7"></a></span>
<span id="cb23-8"><a href="#cb23-8"></a>fitLmPlot &lt;-<span class="st">  </span>sumPlot <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(</span>
<span id="cb23-9"><a href="#cb23-9"></a>    <span class="dt">data =</span> data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">fit=</span>sumMeanPepMod<span class="op">$</span>fitted.values),</span>
<span id="cb23-10"><a href="#cb23-10"></a>    <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x=</span>peptide, <span class="dt">y=</span>fit,<span class="dt">color=</span>condition, <span class="dt">group=</span>sample)) <span class="op">+</span></span>
<span id="cb23-11"><a href="#cb23-11"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;fit: ~ sample + peptide&quot;</span>)</span>
<span id="cb23-12"><a href="#cb23-12"></a>sumLmPlot &lt;-<span class="st"> </span>sumPlot <span class="op">+</span><span class="st"> </span><span class="kw">geom_hline</span>(</span>
<span id="cb23-13"><a href="#cb23-13"></a>    <span class="dt">data =</span> sumMeanPep,</span>
<span id="cb23-14"><a href="#cb23-14"></a>    <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">yintercept=</span>intensity,<span class="dt">color=</span>condition)) <span class="op">+</span></span>
<span id="cb23-15"><a href="#cb23-15"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Summarization: sample effect&quot;</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="kw">grid.arrange</span>(sumMedianPlot, sumLmPlot, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_hline).</code></pre>
<p><img src="pda_robustSummarisation_peptideModels_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<ul>
<li><p>By correcting for the peptide species the protein expression values are much better separated an better reflect differences in abundance induced by the spike-in condition.</p></li>
<li><p>Indeed, it shows that median summarisation that does not account for the peptide effect indeed overestimated the protein expression value in the small spike-in conditions and underestimated that in the large spike-in conditions.</p></li>
<li><p>Still there seem to be some issues with samples that for which the expression values are not well separated according to the spike-in condition.</p></li>
</ul>
<p>A residual analysis clearly indicates potential issues:</p>
<details>
<summary> Click to see code to make plot </summary>
<p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>resPlot &lt;-<span class="st"> </span>data <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb26-2"><a href="#cb26-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">res=</span>sumMeanPepMod<span class="op">$</span>residuals) <span class="op">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> peptide, <span class="dt">y =</span> res, <span class="dt">color =</span> condition, <span class="dt">label =</span> condition), <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb26-4"><a href="#cb26-4"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">shape=</span><span class="dv">21</span>) <span class="op">+</span></span>
<span id="cb26-5"><a href="#cb26-5"></a><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span></span>
<span id="cb26-6"><a href="#cb26-6"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust =</span> <span class="dv">1</span>)) <span class="op">+</span></span>
<span id="cb26-7"><a href="#cb26-7"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Peptide&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb26-8"><a href="#cb26-8"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;residual&quot;</span>) <span class="op">+</span></span>
<span id="cb26-9"><a href="#cb26-9"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;residuals: ~ sample + peptide&quot;</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a><span class="kw">grid.arrange</span>(fitLmPlot, resPlot, <span class="dt">nrow =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="pda_robustSummarisation_peptideModels_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a><span class="kw">grid.arrange</span>(fitLmPlot, sumLmPlot, <span class="dt">nrow =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="pda_robustSummarisation_peptideModels_files/figure-html/unnamed-chunk-22-2.png" width="672" /></p>
<ul>
<li>The residual plot shows some large outliers for peptide KIEEIAAK.</li>
<li>Indeed, in the original plot the intensities for this peptide do not seem to line up very well with the concentration.</li>
<li>This induces a bias in the summarization for some of the samples (e.g. for D and E)</li>
</ul>
</div>
<div id="robust-summarization-using-a-peptide-level-linear-model" class="section level3">
<h3><span class="header-section-number">2.1.3</span> Robust summarization using a peptide-level linear model</h3>
<p><span class="math display">\[ 
y_{ip} = \beta_i^\text{sample}+\beta^{peptide}_{p} + \epsilon_{ip}
\]</span></p>
<ul>
<li>Ordinary least squares: estimate <span class="math inline">\(\beta\)</span> that minimizes <span class="math display">\[
\text{OLS}: \sum\limits_{i,p} \epsilon_{ip}^2 = \sum\limits_{i,p}(y_{ip}-\beta_i^\text{sample}-\beta_p^\text{peptide})^2
\]</span></li>
</ul>
<p>We replace OLS by M-estimation with loss function <span class="math display">\[
\text{OLS}: \sum\limits_{i,p} w_{ip}\epsilon_{ip}^2 = \sum\limits_{i,p}w_{ip}(y_{ip}-\beta_i^\text{sample}-\beta_p^\text{peptide})^2
\]</span></p>
<ul>
<li>Iteratively fit model with observation weights <span class="math inline">\(w_{ip}\)</span> until convergence</li>
<li>The weights are calculated based on standardized residuals</li>
</ul>
<details>
<summary> Click to see code to make plot </summary>
<p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>sumMeanPepRobMod &lt;-<span class="st"> </span>MASS<span class="op">::</span><span class="kw">rlm</span>(intensity <span class="op">~</span><span class="st"> </span><span class="dv">-1</span> <span class="op">+</span><span class="st"> </span>sample <span class="op">+</span><span class="st"> </span>peptide,data)</span>
<span id="cb29-2"><a href="#cb29-2"></a>resRobPlot &lt;-<span class="st"> </span>data <span class="op">%&gt;%</span></span>
<span id="cb29-3"><a href="#cb29-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">res =</span> sumMeanPepRobMod<span class="op">$</span>residuals,</span>
<span id="cb29-4"><a href="#cb29-4"></a>         <span class="dt">w =</span> sumMeanPepRobMod<span class="op">$</span>w) <span class="op">%&gt;%</span></span>
<span id="cb29-5"><a href="#cb29-5"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> peptide, <span class="dt">y =</span> res, <span class="dt">color =</span> condition, <span class="dt">label =</span> condition,<span class="dt">size=</span>w), <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></span>
<span id="cb29-6"><a href="#cb29-6"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">shape=</span><span class="dv">21</span>,<span class="dt">size=</span>.<span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb29-7"><a href="#cb29-7"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">shape=</span><span class="dv">21</span>) <span class="op">+</span></span>
<span id="cb29-8"><a href="#cb29-8"></a><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span></span>
<span id="cb29-9"><a href="#cb29-9"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust =</span> <span class="dv">1</span>)) <span class="op">+</span></span>
<span id="cb29-10"><a href="#cb29-10"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Peptide&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb29-11"><a href="#cb29-11"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;residual&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb29-12"><a href="#cb29-12"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>)<span class="op">*</span><span class="kw">max</span>(<span class="kw">abs</span>(sumMeanPepRobMod<span class="op">$</span>residuals)))</span>
<span id="cb29-13"><a href="#cb29-13"></a>weightPlot &lt;-<span class="st"> </span><span class="kw">qplot</span>(</span>
<span id="cb29-14"><a href="#cb29-14"></a>  <span class="kw">seq</span>(<span class="op">-</span><span class="dv">5</span>,<span class="dv">5</span>,.<span class="dv">01</span>), </span>
<span id="cb29-15"><a href="#cb29-15"></a>  MASS<span class="op">::</span><span class="kw">psi.huber</span>(<span class="kw">seq</span>(<span class="op">-</span><span class="dv">5</span>,<span class="dv">5</span>,.<span class="dv">01</span>)),</span>
<span id="cb29-16"><a href="#cb29-16"></a>  <span class="dt">geom=</span><span class="st">&quot;path&quot;</span>) <span class="op">+</span></span>
<span id="cb29-17"><a href="#cb29-17"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;standardized residual&quot;</span>) <span class="op">+</span></span>
<span id="cb29-18"><a href="#cb29-18"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;weight&quot;</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="kw">grid.arrange</span>(weightPlot,resRobPlot,<span class="dt">nrow=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="pda_robustSummarisation_peptideModels_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<ul>
<li>We clearly see that the weights in the M-estimation procedure will down-weight errors associated with outliers for peptide KIEEIAAK.</li>
</ul>
<details>
<summary> Click to see code to make plot </summary>
<p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>sumMeanPepRob &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</span>
<span id="cb31-2"><a href="#cb31-2"></a>  <span class="dt">intensity=</span>sumMeanPepRobMod<span class="op">$</span>coef[<span class="kw">grep</span>(<span class="st">&quot;sample&quot;</span>,<span class="kw">names</span>(sumMeanPepRobMod<span class="op">$</span>coef))] <span class="op">+</span><span class="st"> </span><span class="kw">mean</span>(data<span class="op">$</span>intensity) <span class="op">-</span><span class="st"> </span><span class="kw">mean</span>(sumMeanPepRobMod<span class="op">$</span>coef[<span class="kw">grep</span>(<span class="st">&quot;sample&quot;</span>,<span class="kw">names</span>(sumMeanPepRobMod<span class="op">$</span>coef))]),</span>
<span id="cb31-3"><a href="#cb31-3"></a>  <span class="dt">condition=</span> <span class="kw">names</span>(sumMeanPepRobMod<span class="op">$</span>coef)[<span class="kw">grep</span>(<span class="st">&quot;sample&quot;</span>,<span class="kw">names</span>(sumMeanPepRobMod<span class="op">$</span>coef))] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">substr</span>(<span class="dv">18</span>,<span class="dv">18</span>) <span class="op">%&gt;%</span><span class="st"> </span>as.factor )</span>
<span id="cb31-4"><a href="#cb31-4"></a></span>
<span id="cb31-5"><a href="#cb31-5"></a>sumRlmPlot &lt;-<span class="st"> </span>sumPlot <span class="op">+</span><span class="st"> </span><span class="kw">geom_hline</span>(</span>
<span id="cb31-6"><a href="#cb31-6"></a>    <span class="dt">data=</span>sumMeanPepRob,</span>
<span id="cb31-7"><a href="#cb31-7"></a>    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">yintercept=</span>intensity,<span class="dt">color=</span>condition)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb31-8"><a href="#cb31-8"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Robust&quot;</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a> <span class="kw">grid.arrange</span>(sumLmPlot <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;OLS&quot;</span>), sumRlmPlot, <span class="dt">nrow =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="pda_robustSummarisation_peptideModels_files/figure-html/unnamed-chunk-26-1.png" width="672" /> - Robust regresion results in a better separation between the protein expression values for the different samples according to their spike-in concentration.</p>
</div>
<div id="maxlfq-summarization" class="section level3">
<h3><span class="header-section-number">2.1.4</span> maxLFQ summarization</h3>
<p><img src="figures/maxLFQ_principle.png" /><!-- --></p>
</div>
<div id="ms-stats-summarization" class="section level3">
<h3><span class="header-section-number">2.1.5</span> MS-stats summarization</h3>
<ul>
<li>MS-stats also uses a robust peptide level model to perform the summarization, however, they typically first impute missing values</li>
</ul>
</div>
<div id="comparison-summarization-methods" class="section level3">
<h3><span class="header-section-number">2.1.6</span> Comparison summarization methods</h3>
<p><img src="figures/maxLFQ_principle.png" /><!-- --></p>
<ul>
<li><span class="citation">(Sticker et al. 2020)</span></li>
<li>doi: <a href="https://doi.org/10.1074/mcp.RA119.001624" class="uri">https://doi.org/10.1074/mcp.RA119.001624</a><br />
</li>
<li><a href="https://www.mcponline.org/action/showPdf?pii=S1535-9476%2820%2934982-3">pdf</a></li>
</ul>
</div>
</div>
<div id="estimation-of-differential-abundance-using-peptide-level-model" class="section level2">
<h2><span class="header-section-number">2.2</span> Estimation of differential abundance using peptide level model</h2>
<ul>
<li>Instead of summarising the data we can also directly model the data at the peptide-level.</li>
<li>But, we will have to address the pseudo-replication.</li>
</ul>
<p><span class="math display">\[
y_{iclp}= \beta_0 + \beta_c^\text{condition} + \beta_l^\text{lab} + \beta_p^\text{peptide} + u_s^\text{sample} + \epsilon_{iclp}
\]</span></p>
<ul>
<li><p>protein-level</p>
<ul>
<li><span class="math inline">\(\beta^\text{condition}_c\)</span>: spike-in condition</li>
<li><span class="math inline">\(\beta^\text{condition}_c\)</span>: lab effect</li>
<li><span class="math inline">\(u_{r}^\text{run}\sim N\left(0,\sigma^2_\text{run}\right) \rightarrow\)</span> random effect addresses pseudo-replication</li>
</ul></li>
<li><p>peptide-level</p>
<ul>
<li><span class="math inline">\(\beta_{p}^\text{peptide}\)</span>: peptide effect</li>
<li><span class="math inline">\(\epsilon_{rp} \sim N\left(0,\sigma^2_{\epsilon}\right)\)</span> within sample (run) error</li>
</ul></li>
<li><p>DA estimates: <span class="math display">\[
\log_2FC_{B-A}=\beta^\text{condition}_B  
\]</span> <span class="math display">\[
\log_2FC_{C-B}=\beta^\text{condition}_C - \beta^\text{condition}_B 
\]</span></p></li>
<li><p>Mixed peptide-level models are implemented in msqrob2</p></li>
<li><p>It has the advantages that</p>
<ol style="list-style-type: decimal">
<li>it correctly addresses the difference levels of variability in the data</li>
<li>it avoids summarisation and therefore also accounts for the difference in the number of peptides that are observed in each sample</li>
<li>more powerful analysis</li>
</ol></li>
<li><p>It has the disadvantage that</p>
<ol style="list-style-type: decimal">
<li>protein summaries are no longer available for plotting</li>
<li>it is difficult to correctly specify the degrees of freedom for the test-statistic leading to inference that is too liberal in experiments with small sample size</li>
<li>sometimes sample level random effect variance are estimated to be zero, then the pseudo-replication is not addressed leading to inference that is too liberal for these specific proteins</li>
<li>they are much more difficult to disseminate to users with limited background in statistics</li>
</ol></li>
</ul>
<p>Hence, for this course we opted to use peptide-level models for summarization, but not for directly inferring on the differential expression at the protein-level.</p>
<div id="refs" class="references">
<div id="ref-sticker2020">
<p>Sticker, A., L. Goeminne, L. Martens, and L. Clement. 2020. “Robust Summarization and Inference in Proteome-wide Label-free Quantification.” <em>Mol Cell Proteomics</em> 19 (7): 1209–19.</p>
</div>
</div>
</div>
</div>

<div id="rmd-source-code">---
title: "Statistical Methods for Quantitative MS-based Proteomics: Peptide-level Models for Summarization and Inference"
author: "Lieven Clement"
date: "[statOmics](https://statomics.github.io), Ghent University"
output:
    html_document:
      code_download: true
      theme: cosmo
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
    pdf_document:
      toc: true
      number_sections: true
linkcolor: blue
urlcolor: blue
citecolor: blue

bibliography: msqrob2.bib
      
---

<a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a>

This is part of the online course [Proteomics Data Analysis 2021 (PDA21)](https://statomics.github.io/PDA21/)

# Import the data in R 

## Load libraries 

<details><summary> Click to see code </summary><p>
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(QFeatures)
library(msqrob2)
library(plotly)
library(ggplot2)
library(gridExtra)
```
</p></details>

## Read data 

<details><summary> Click to see background and code </summary><p>
1. We use a peptides.txt file from MS-data quantified with maxquant that 
contains MS1 intensities summarized at the peptide level. 
```{r}
peptidesFile <- "https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/fullCptacDatasSetNotForTutorial/peptides.txt"
```

2. Maxquant stores the intensity data for the different samples in columnns that start with Intensity. We can retreive the column names with the intensity data with the code below: 

```{r}
ecols <- grep("Intensity\\.", names(read.delim(peptidesFile)))
```

3. Read the data and store it in  QFeatures object 

```{r}
pe <- readQFeatures(
  table = peptidesFile,
  fnames = 1,
  ecol = ecols,
  name = "peptideRaw", sep="\t")
```
</p></details>

## Design

<details><summary> Click to see background and code </summary><p>

```{r} 
pe %>% colnames
```

- Note, that the sample names include the spike-in condition. 
- They also end on a number. 
  
  - 1-3 is from lab 1, 
  - 4-6 from lab 2 and 
  - 7-9 from lab 3. 

- We update the colData with information on the design

```{r}
colData(pe)$lab <- rep(rep(paste0("lab",1:3),each=3),5) %>% as.factor
colData(pe)$condition <- pe[["peptideRaw"]] %>% colnames %>% substr(12,12) %>% as.factor
colData(pe)$spikeConcentration <- rep(c(A = 0.25, B = 0.74, C = 2.22, D = 6.67, E = 20),each = 9)
```

- We explore the colData

```{r}
colData(pe)
```

</p></details>

## Preprocessing

### Log-transform

<details><summary> Click to see code to log-transfrom the data </summary><p>
- We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.

```{r}
rowData(pe[["peptideRaw"]])$nNonZero <- rowSums(assay(pe[["peptideRaw"]]) > 0)
```


- Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.

```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA
```

- Logtransform data with base 2

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")
```
</p></details>


### Filtering

<details><summary> Click to see code to filter the data </summary><p>

1. Handling overlapping protein groups

In our approach a peptide can map to multiple proteins, as long as there is
none of these proteins present in a smaller subgroup.

```{r}
pe[["peptideLog"]] <-
 pe[["peptideLog"]][rowData(pe[["peptideLog"]])$Proteins
 %in% smallestUniqueGroups(rowData(pe[["peptideLog"]])$Proteins),]
```

2. Remove reverse sequences (decoys) and contaminants

We now remove the contaminants, peptides that map to decoy sequences, and proteins
which were only identified by peptides with modifications.

```{r}
pe[["peptideLog"]] <- pe[["peptideLog"]][rowData(pe[["peptideLog"]])$Reverse != "+", ]
pe[["peptideLog"]] <- pe[["peptideLog"]][rowData(pe[["peptideLog"]])$
    Potential.contaminant != "+", ]
```

3. Drop peptides that were only identified in one sample

We keep peptides that were observed at last twice.

```{r}
pe[["peptideLog"]] <- pe[["peptideLog"]][rowData(pe[["peptideLog"]])$nNonZero >= 2, ]
nrow(pe[["peptideLog"]])
```

We keep `r nrow(pe[["peptideLog"]])` peptides upon filtering.
</p></details>

## Normalization 

<details><summary> Click to see R-code to normalize the data </summary><p>
```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```
</p></details>

---

# Peptide-level models 

## Summarization 


<details><summary> Click to see code to make plot </summary><p>
```{r plot = FALSE}

prot <- "P01031ups|CO5_HUMAN_UPS"
data <- pe[["peptideNorm"]][
  rowData(pe[["peptideNorm"]])$Proteins == prot,
  colData(pe)$lab=="lab3"] %>%
  assay %>%
  as.data.frame %>%
  rownames_to_column(var = "peptide") %>%
  gather(sample, intensity, -peptide) %>% 
  mutate(condition = colData(pe)[sample,"condition"]) %>%
  na.exclude
sumPlot <- data %>%
  ggplot(aes(x = peptide, y = intensity, color = condition, group = sample, label = condition), show.legend = FALSE) +
  geom_text(show.legend = FALSE) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Peptide") + 
  ylab("Intensity (log2)") +
  ggtitle(paste0("protein: ",prot))
```
</p></details>


Here, we will focus on the summarization of the intensities for protein `r prot`.

```{r}
sumPlot
```

### Median summarization

We first evaluate median summarization for protein `r prot`.

<details><summary> Click to see code to make plot </summary><p>
```{r}
dataHlp <- pe[["peptideNorm"]][
    rowData(pe[["peptideNorm"]])$Proteins == prot,
    colData(pe)$lab=="lab3"] %>% assay 

sumMedian <- data.frame(
  intensity= dataHlp
    %>% colMedians(na.rm=TRUE)
  ,
  condition= colnames(dataHlp) %>% substr(12,12) %>% as.factor )

sumMedianPlot <- sumPlot + 
  geom_hline(
    data = sumMedian,
    mapping = aes(yintercept=intensity,color=condition)) + 
  ggtitle("Median summarization")
```
</p></details>

```{r}
sumMedianPlot
```


- The sample medians are not a good estimate for the protein expression value. 
- Indeed, they do not account for differences in peptide effects
- Peptides that ionize poorly are also picked up in samples with high spike-in concencentration and not in samples with low spike-in concentration
- This introduces a bias. 

### Model based summarization

We can use a linear peptide-level model to estimate the protein expression value while correcting for the peptide effect, i.e. 

$$ 
y_{ip} = \beta_i^\text{sample}+\beta^{peptide}_{p} + \epsilon_{ip}
$$


<details><summary> Click to see code to make plot </summary><p>
```{r}
sumMeanPepMod <- lm(intensity ~ -1 + sample + peptide,data)

sumMeanPep <- data.frame(
  intensity=sumMeanPepMod$coef[grep("sample",names(sumMeanPepMod$coef))] + mean(data$intensity) - mean(sumMeanPepMod$coef[grep("sample",names(sumMeanPepMod$coef))]),
  condition= names(sumMeanPepMod$coef)[grep("sample",names(sumMeanPepMod$coef))] %>% substr(18,18) %>% as.factor )


fitLmPlot <-  sumPlot + geom_line(
    data = data %>% mutate(fit=sumMeanPepMod$fitted.values),
    mapping = aes(x=peptide, y=fit,color=condition, group=sample)) +
    ggtitle("fit: ~ sample + peptide")
sumLmPlot <- sumPlot + geom_hline(
    data = sumMeanPep,
    mapping = aes(yintercept=intensity,color=condition)) +
    ggtitle("Summarization: sample effect")
```
</p></details>

```{r}
grid.arrange(sumMedianPlot, sumLmPlot, ncol=2)
```

- By correcting for the peptide species the protein expression values are much better separated an better reflect differences in abundance induced by the spike-in condition. 

- Indeed, it shows that median summarisation that does not account for the peptide effect indeed overestimated the protein expression value in the small spike-in conditions and underestimated that in the large spike-in conditions.

- Still there seem to be some issues with samples that for which the expression values are not well separated according to the spike-in condition. 

A residual analysis clearly indicates potential issues:

<details><summary> Click to see code to make plot </summary><p>
```{r}
resPlot <- data %>% 
  mutate(res=sumMeanPepMod$residuals) %>%
  ggplot(aes(x = peptide, y = res, color = condition, label = condition), show.legend = FALSE) +
  geom_point(shape=21) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Peptide") + 
  ylab("residual") +
  ggtitle("residuals: ~ sample + peptide")
```
</p></details>

```{r}
grid.arrange(fitLmPlot, resPlot, nrow = 1)
grid.arrange(fitLmPlot, sumLmPlot, nrow = 1)
```

- The residual plot shows some large outliers for peptide KIEEIAAK. 
- Indeed, in the original plot the intensities for this peptide do not seem to line up very well with the concentration. 
- This induces a bias in the summarization for some of the samples (e.g. for D and E)

### Robust summarization using a peptide-level linear model 

$$ 
y_{ip} = \beta_i^\text{sample}+\beta^{peptide}_{p} + \epsilon_{ip}
$$


- Ordinary least squares: estimate $\beta$ that minimizes
$$
\text{OLS}: \sum\limits_{i,p} \epsilon_{ip}^2 = \sum\limits_{i,p}(y_{ip}-\beta_i^\text{sample}-\beta_p^\text{peptide})^2
$$

We replace OLS by M-estimation with loss function
$$
\text{OLS}: \sum\limits_{i,p} w_{ip}\epsilon_{ip}^2 = \sum\limits_{i,p}w_{ip}(y_{ip}-\beta_i^\text{sample}-\beta_p^\text{peptide})^2
$$

- Iteratively fit model with observation weights $w_{ip}$ until convergence
- The weights are calculated based on standardized residuals

<details><summary> Click to see code to make plot </summary><p>
```{r}
sumMeanPepRobMod <- MASS::rlm(intensity ~ -1 + sample + peptide,data)
resRobPlot <- data %>%
  mutate(res = sumMeanPepRobMod$residuals,
         w = sumMeanPepRobMod$w) %>%
  ggplot(aes(x = peptide, y = res, color = condition, label = condition,size=w), show.legend = FALSE) +
  geom_point(shape=21,size=.2) +
  geom_point(shape=21) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Peptide") + 
  ylab("residual") + 
  ylim(c(-1,1)*max(abs(sumMeanPepRobMod$residuals)))
weightPlot <- qplot(
  seq(-5,5,.01), 
  MASS::psi.huber(seq(-5,5,.01)),
  geom="path") +
  xlab("standardized residual") +
  ylab("weight")
```
</p></details>

```{r}
grid.arrange(weightPlot,resRobPlot,nrow=1)
```

- We clearly see that the weights in the M-estimation procedure will down-weight errors associated with outliers for peptide KIEEIAAK.

<details><summary> Click to see code to make plot </summary><p>
```{r}
sumMeanPepRob <- data.frame(
  intensity=sumMeanPepRobMod$coef[grep("sample",names(sumMeanPepRobMod$coef))] + mean(data$intensity) - mean(sumMeanPepRobMod$coef[grep("sample",names(sumMeanPepRobMod$coef))]),
  condition= names(sumMeanPepRobMod$coef)[grep("sample",names(sumMeanPepRobMod$coef))] %>% substr(18,18) %>% as.factor )

sumRlmPlot <- sumPlot + geom_hline(
    data=sumMeanPepRob,
    mapping=aes(yintercept=intensity,color=condition)) + 
    ggtitle("Robust")
```
</p></details>

```{r}
 grid.arrange(sumLmPlot + ggtitle("OLS"), sumRlmPlot, nrow = 1)
```
- Robust regresion results in a better separation between the protein expression values for the different samples according to their spike-in concentration. 

### maxLFQ summarization 

```{r echo=FALSE}
knitr::include_graphics("./figures/maxLFQ_principle.png")
```

### MS-stats summarization

- MS-stats also uses a robust peptide level model to perform the summarization, however, they typically first impute missing values

### Comparison summarization methods 

```{r echo=FALSE}
knitr::include_graphics("./figures/maxLFQ_principle.png")
```

- [@sticker2020]
- doi: https://doi.org/10.1074/mcp.RA119.001624  
- [pdf](https://www.mcponline.org/action/showPdf?pii=S1535-9476%2820%2934982-3)

## Estimation of differential abundance using peptide level model

- Instead of summarising the data we can also directly model the data at the peptide-level. 
- But, we will have to address the pseudo-replication.

$$
y_{iclp}= \beta_0 + \beta_c^\text{condition} + \beta_l^\text{lab} + \beta_p^\text{peptide} + u_s^\text{sample} + \epsilon_{iclp}
$$

- protein-level

  - $\beta^\text{condition}_c$: spike-in condition
  - $\beta^\text{condition}_c$: lab effect
  - $u_{r}^\text{run}\sim N\left(0,\sigma^2_\text{run}\right) \rightarrow$ random effect addresses pseudo-replication

- peptide-level 
  - $\beta_{p}^\text{peptide}$: peptide effect
  - $\epsilon_{rp} \sim N\left(0,\sigma^2_{\epsilon}\right)$ within sample (run) error 


- DA estimates: 
$$
\log_2FC_{B-A}=\beta^\text{condition}_B  
$$
$$
\log_2FC_{C-B}=\beta^\text{condition}_C - \beta^\text{condition}_B 
$$
- Mixed peptide-level models are implemented in msqrob2

- It has the advantages that

  1. it correctly addresses the difference levels of variability in the data
  2. it avoids summarisation and therefore also accounts for the difference in the number of peptides that are observed in each sample
  3. more powerful analysis

- It has the disadvantage that

  
  1. protein summaries are no longer available for plotting
  2. it is difficult to correctly specify the degrees of freedom for the test-statistic leading to inference that is too liberal in experiments with small sample size
  3. sometimes sample level random effect variance are estimated to be zero, then the pseudo-replication is not addressed leading to inference that is too liberal for these specific proteins
  4. they are much more difficult to disseminate to users with limited background in statistics

Hence, for this course we opted to use peptide-level models for summarization, but not for directly inferring on the differential expression at the protein-level. 

</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("pda_robustSummarisation_peptideModels.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
